app/models/spree/order_decorator.rb

```diff
@@ -12,29 +14,8 @@ def gift_card_match(line_item, options)
             gc_detail_set.superset?(options_set)
           end
         end
-  
```

Comment:
why are we removing the `finalize!` method from order?

Reply:
because we should not override this and use the state machine to prevent the race condition

Reply:
👍